### PR TITLE
Feat module reference reorganization

### PR DESCRIPTION
**Pull request by JaydenLiang**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
-
-const entries = {};
-config.function.forEach(f => {
-    entries[f.name] = path.resolve(__dirname, f.source.location);
-});
+const path = require('path');
 
 module.exports = {
-    entry: entries,
+    entry: {
```
Comment (jamie-pate) [inline webpack.config.js:8]: I got this working with multiple entries... need to add entries and use `splitChunks` in the optimization section for webpack.

**Review by jamie-pate — CHANGES_REQUESTED** [review]

**Comment by JaydenLiang** [comment]
@jamie-pate adding the entryPoint to each function solves the problem. much simpler. no need to slipt chunks now. thanks for pointing out the entryPoint method.

File: webpack.config.js [hunk 2]
```diff
@@ -45,9 +43,9 @@ module.exports = {
     },
     output: {
         filename: () => {
-            return 'index.js';
+            return './[name]/index-bundle.js';
```
Comment (jamie-pate) [inline webpack.config.js:46]: can reset this now

File: fgt-as-handler/function.json [hunk 3]
```diff
@@ -13,5 +13,6 @@
             "name": "res"
         }
     ],
-    "scriptFile": "../dist/fgt-as-handler/index.js"
+    "scriptFile": "../dist/autoscale-shared/index-bundle.js",
```
Comment (jamie-pate) [inline fgt-as-handler/function.json:16]: just index.js ?

**Review by jamie-pate — CHANGES_REQUESTED** [review]

File: scripts/make-dist.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    'templates',
+    'package.json',
+    'README.md',
+    'LICENSE'
+];
+
+// zip function app first
+console.log(chalk.blueBright('( ͡° ͜ʖ ͡°)'), ' processing function app.');
+// remove the zip file if exists.
+shelljs.rm('-f', fileFuncapp);
```
Comment (jamie-pate) [inline scripts/make-dist.ts:70]: this could be `await rmdir(dirname, {recursive: true})`
`const { promises: { rmdir }} = require('fs');`

**Review by jamie-pate — CHANGES_REQUESTED** [review]

**Review by jamie-pate — APPROVED** [review]

**Pull request merged**